src/main/java/blusunrize/immersiveengineering/client/ClientEventHandler.java

```diff
@@ -1282,6 +1281,59 @@ public void onRenderWorldLastEvent(RenderWorldLastEvent event)
 				ClientUtils.mc().world.getTileEntity(((BlockRayTraceResult)ClientUtils.mc().objectMouseOver).getPos()) instanceof SampleDrillTileEntity)
 			chunkBorders = true;
 
+		if(Screen.hasShiftDown())
+		{
+			Tessellator tessellator = Tessellator.getInstance();
+			BufferBuilder BufferBuilder = tessellator.getBuffer();
+			double px = TileEntityRendererDispatcher.staticPlayerX;
+			double py = TileEntityRendererDispatcher.staticPlayerY;
+			double pz = TileEntityRendererDispatcher.staticPlayerZ;
+			GlStateManager.disableTexture();
+			GlStateManager.enableBlend();
+			GlStateManager.disableCull();
+			GlStateManager.blendFuncSeparate(770, 771, 1, 0);
+			GlStateManager.shadeModel(GL11.GL_SMOOTH);
+			GlStateManager.lineWidth(5f);
+			List<ResourceLocation> keyList = new ArrayList<>(MineralMix.mineralList.keySet());
+			keyList.sort(Comparator.comparing(ResourceLocation::toString));
+			for(MineralVein vein : ExcavatorHandler.getMineralVeinList().get(DimensionType.OVERWORLD))
+			{
+				ColumnPos pos = vein.getPos();
+				int iC = keyList.indexOf(vein.getMineral().getId());
+				DyeColor color = DyeColor.values()[iC%16];
+				float[] rgb = color.getColorComponentValues();
+				float r = rgb[0];
+				float g = rgb[1];
+				float b = rgb[2];
+
+				BufferBuilder.setTranslation(pos.x-px, 0-py, pos.z-pz);
+				BufferBuilder.begin(GL11.GL_LINES, DefaultVertexFormats.POSITION_COLOR);
+				BufferBuilder.pos(0, 0, 0).color(r, g, b, .75f).endVertex();
+				BufferBuilder.pos(0, 128, 0).color(r, g, b, .75f).endVertex();
+				tessellator.draw();
+
+				int radius = vein.getRadius();
+				float angle;
+				double x1;
+				double z1;
+				BufferBuilder.begin(GL11.GL_LINE_LOOP, DefaultVertexFormats.POSITION_COLOR);
+				for(int p = 0; p < 12; p++)
+				{
+					angle = 360.0f/12*p;
+					x1 = radius*Math.cos(angle*Math.PI/180);
+					z1 = radius*Math.sin(angle*Math.PI/180);
+					BufferBuilder.pos(x1, 70, z1).color(r, g, b, .75f).endVertex();
+				}
+				tessellator.draw();
+
+				BufferBuilder.setTranslation(0, 0, 0);
+			}
+			GlStateManager.shadeModel(GL11.GL_FLAT);
+			GlStateManager.enableCull();
+			GlStateManager.disableBlend();
+			GlStateManager.enableTexture();
+		}
+
```

Comment:
I assume this is the debug code that needs to be remove/disabled?

Reply:
Yup.